vb-be/src/modules/room/room.schema.ts

```diff
@@ -10,19 +10,11 @@ export interface RoomType {
   start?: Date;
   end?: Date;
   maxUsers?: number;
-  currentUsers?: UserType;
+  currentUsers?: UserType[];
 }
 
 export interface IRoom extends Document {
-  id: string;
-  name: string;
-  description: string;
-  tags: TagType[];
-  host: string;
-  start: Date;
-  end: Date;
-  maxUsers?: number;
-  currentUsers?: UserType;
+  _id?: Types.ObjectId;
```

Comment:
Curious, why make `_id` optional here when _id is a for-sure field for the `Room` object?  I feel like implementing this across the board will make handling data harder in the future, in cases where we're needing `_id` and all of the objects down the line are utilizing the optional ID type like this

Reply:
because if you see the `Document` structure of Mongo, its `_id?` which is its optional. I was just keeping it the same as that

Reply:
That's super weird, so we can't ever guarantee the ID value?

Reply:
i have no idea lol, i'm just following the what mongoose has. but even if we make it required, it should be fine because that object will always have the `_id`